torch/csrc/jit/codegen/cuda/kernel_ir.cpp

```diff
@@ -328,16 +336,35 @@ void Scope::insert_after(Expr* ref, Expr* expr) {
       " after the reference: ",
       ref,
       " however the reference was not found in this scope.");
-  exprs_.insert(it + 1, expr);
+  insert(it + 1, expr);
+}
+
+void Scope::insert(size_t pos, Expr* expr) {
+  const auto it = exprs_.begin() + pos;
+  insert(it, expr);
+}
+
+void Scope::erase(std::vector<Expr*>::const_iterator pos) {
+  // Remove the scope of the expr if this is the scope
+  auto expr = *pos;
+  if (expr->scope() == this) {
+    expr->removeScope();
+  }
```

Comment:
Else code path deserves assert as it shouldn't be case.

Reply:
Fixed. Thanks for the feedback. It was actually happening, but that itself was a problematic usage of this interface.